### ISSUE TITLE
Require for detailed tutorial

### ISSUE THREAD

**Issue by ruarua6666** [VT-NLP/Mocheg#4]
Hi, Barry Menglong Yao,

First and foremost, I'd like to express my admiration for the incredible work you've done on Mocheg. It's evident that a lot of effort and dedication went into it, and I truly appreciate your contribution to the community.

I am genuinely interested in diving deeper into your project and leveraging it for my own research/work. However, I've encountered some challenges in setting everything up. Would it be possible for you to provide a more detailed tutorial? Specifically, clarity on the data set storage path, the pre-trained model storage path, and any other configuration details would be extremely beneficial.

Understanding that you are busy, I hope this isn't too much of an inconvenience. Your guidance would significantly assist many of us in the community who are keen to make the most out of your wonderful project.

Thank you so much for considering my request, and once again, hats off to your outstanding work!

Warm regards.

**Comment by Barry-Menglong-Yao** [comment]
Thanks for your interest! It will be helpful to add a detailed tutorial. I will work on it.

In order to solve the challenges in your case, do you mind listing these challenges?

**Comment by ruarua6666** [comment]
Thank you for your time! 
I would like to know how I should place the dataset in the folder? For example, should I create a folder called "data" and put all the files in the dataset in it?

**Comment by Barry-Menglong-Yao** [comment]
Yes. A folder named "data" for the dataset and a folder named "checkpoint" for checkpoints. 

In fact, you can also use other paths as long as you pass your corresponding paths as Python arguments. They are Python arguments defined in corresponding scripts like retrieve_train.py.

Please list other challenges so others can also benefit from your experience.

**Comment by ruarua6666** [comment]
Thank you for your patience and help.
I'm at an academic conference right now and can't start the experiment right away, I'll be coming in next week to start the experiment and I'll list the challenges I'm having then.

**Comment by ruarua6666** [comment]
I encountered an error when running python retrieve_train.py --train_config=BI_ENCODER. The error occurred at the loss_value = loss_model(features, labels) statement in /Mocheg-main/retrieval/training/model.py. The error message is RuntimeError: The size of tensor a (89) must match the size of tensor b (77) at non-singleton dimension 1, and the size of tensor a varies each time I run it. Do you know why this might be happening, and how I could resolve it? Thank you.

**Comment by ruarua6666** [comment]
When I'm running `python retrieve_train.py --train_config=BI_ENCODER`, I encounter an error at the line `loss_value = loss_model(features, labels)` in `/Mocheg-main/retrieval/training/model.py`. The error message is: `RuntimeError: The size of tensor a (89) must match the size of tensor b (77) at non-singleton dimension 1`. Additionally, the size of tensor ‘a’ varies each time I run the code. Do you know why this might be happening and how I can resolve it?

**Comment by ruarua6666** [comment]
Hello,

When I executed the command 'python verify.py --model_type=CLAIM_TEXT_IMAGE_attention_5_4 --batch_size=2048 --lr=0.001 --loss_weight_power=2', I encountered an error in the 'Mocheg-main/verification/train_verify.py' at the line 'output = model( claim_img_encod,device)':

RuntimeError: The size of tensor a (85) must match the size of tensor b (77) at non-singleton dimension 1.

Upon further inspection, I noticed that the issue might be due to the absence of a truncation length in the self.processor function within the /Mocheg-main/verification/util/data_util.py file:

self.processor(text=claim_and_text_evidence_list, images=img_list, return_tensors="pt", padding=True,truncation=True)

When I set the max_length parameter to 77, the code ran without any errors. I'd like to inquire if my solution is appropriate or if there might be an issue with the model I loaded?

Thank you for your assistance!

**Comment by ruarua6666** [comment]
Hi,

While running the command python verify.py --model_type=CLAIM_TEXT_IMAGE_attention_5_4 --batch_size=2048 --lr=0.001 --loss_weight_power=2, I encountered an error in Mocheg-main/verification/train_verify.py at the line output = model(claim_img_encod, device):

RuntimeError: The size of tensor a (85) must match the size of tensor b (77) at non-singleton dimension 1.

I traced the issue back to the file /Mocheg-main/verification/util/data_util.py where the line self.processor(text=claim_and_text_evidence_list, images=img_list, return_tensors="pt", padding=True, truncation=True, max_length=77) did not specify a truncation length. After setting max_length to 77, the code ran without issues.

Is this the correct solution? Or is there an issue with the model I'm loading?

Thank you for your assistance!

**Comment by Barry-Menglong-Yao** [comment]
Yes. The key is to truncate the text under 77 tokens. 

It seems I truncated the text during another preprocess. But your solution is better. I have merged your modification into the repository. Thanks!

**Comment by OPilgrim** [comment]
After I trained the code for the generation module, I removed the `--do_train` option and kept only `--do_predict`. and loaded the model and tokenizer under the `/data/Projects/Mocheg/controllable/generation/output/bart/run_40/without/` path for inference. The generated result was completely different from the generated result just after training. For example,
```
Just after training with --do_train --do_predict setting:
species of grasshopper, the Dactylotum bicolor species has a colorful, bright-red coloration that is similar to the grasshoppers found in other types of bugs, such as the painted woodhopper (Scolopendra subspinipes) and the painted water bug (Anodorhynchus sanguineus). According to the Inaturalist, this species is a member of the family Vespidae, one of the few species that is endemic to North America. It is native to a few locations in North America (primarily in the eastern half of the United States), and is found in southern Mexico, Central America, and the Caribbean. The web site for the species notes that it is common in some parts of the world, particularly in central and eastern Europe. 'It is found on some of the central African countries such as South Africa, Uganda, and Zambia,' wrote the INaturalist. Furthermore, the photograph displayed above was not the only one of its kind spotted in Michigan in September 2021. Here's a look at a photograph of the genuine rainbow grasshoppe as published by the Reddit page Nature Is FuckingLit: Given that the species shown above is different, but the colors in both the photograph and the written description are similar, we rate this claim as a 'Mixture,' as well as the possibility that it could be a different species of Grasshopper species found in different locations, as of this writing, as 'Unproven.' We reached out to the Smithsonian, and will update this post if more information becomes available. As such, the rating of this claim is 'Miscaptioned,' with the status changed to 'False.'

Inference with only  --do_predict setting:
NEI  A photograph shared to Reddit in September 2021 showed a vibrantly hued grasshopper that was seen in Michigan.  Sources:  dosangst. '🔥 Really Cool Grasshopper Spotted in a Backyard in S.E. Michigan.' R/NatureIsFuckingLit, 12 Sept. 2021, [www.reddit.com/r/Nature_Is_Fucking_Lit/comments/pmy84s/really_cool_grasshopper_spotted_in_a_backyard_in/](https://github.com/VT-NLP/Mocheg/issues/www.reddit.com/r/Nature_Is_Fucking_Lit/comments/pmy84s/really_cool_grasshopper_spotted_in_a_backyard_in/). 'Is This a Real Rainbow Grasshoppers?' Snopes.Com, https://www.snopes.net/node/view/31526. Accessed 16 Sept. 2011. Species Dactylotum Bicolor. SCAN Collections Search Results. https://scan-bugs.org/portal/collections/list.php?portal_id=1&type=2&db=all&page=1. Accessed 15 Sept. 2016. 'Dactylotsum bicolor'. Bug Guide. https/bugguide.org?portals=1 &type=3&db =1&page_id =1 &page_type=1 and https://bugguide-org.org/?type=4&db_id=-1&site=1-1&search=1_1_2_1&index=2. Accessed 17 Sept. 2017. 'Rainbow Grasshoper (Dactyotum biculatus).' INaturalist.com. https//[www.inaturalism.com/?article=rainbow-grasshoppers&id=179324&article_url=http%3A%2F%2B%2C%2A%22inaturalist%2D%3C%3B%3F%3D%2E%3E%](https://github.com/VT-NLP/Mocheg/issues/www.inaturalism.com/?article=rainbow-grasshoppers&id=179324&article_url=http%3A%2F%2B%2C%2A%22inaturalist%2D%3C%3B%3F%3D%2E%3E%)
```

I also loaded all checkpoints in the `without/` path, and all generated the same result. So I was confused about how I was supposed to load the trained model to get the perfect result.
![image](https://github.com/VT-NLP/Mocheg/assets/24997865/777f6cdc-2bf2-4fac-a008-0a3a66c3b064)
